agent-inject/handler.go

```diff
@@ -245,6 +265,13 @@ func (h *Handler) Mutate(req *admissionv1.AdmissionRequest) *admissionv1.Admissi
 		return admissionError(req.UID, err)
 	}
 
+	// Expose when workloads are injecting initContainers only. This is useful to identify
+	// workloads that are likely good candidates for migration to Vault Secrets Operator for
+	// greater scalability and performance.
+	if agentSidecar.PrePopulateOnly {
+		resp.Warnings = append(resp.Warnings, warningInitOnlyInjection)
+	}
```

Comment:
Instead of adding onto the Warning that's sent back to the user here, I'd suggest changing Mutate() to return a struct that contains the AdmissionResponse along with metadata fields for what was injected, i.e. a couple of bool's for `init` and `sidecar` injections. Something like:

```go
type MutateResponse struct {
	InjectedInit    bool
	InjectedSidecar bool
	Resp            *admissionv1.AdmissionResponse
}
```

That would also make it easier for the metrics to show the other injection case of sidecar only.

Alternately we could perhaps set the injected container info in the AuditAnnotations map on the AdmissionResponse.

Reply:
Sounds good. I looked into AuditAnnotations, but since those would get persisted to Kubernetes audit logs decided to go with Warnings instead.

Agreed that it's better to just change the signature of `Mutate()` and return the metadata we need for telemetry.